app/colorCoder.js

```diff
@@ -1,99 +1,47 @@
-	pair.minorColor = MinorColorNames[minorIndex];
+	let majorIndex = parseInt(zeroBasedPairNumber / minorSize);
+	let minorIndex = parseInt(zeroBasedPairNumber % minorSize);
+	let pair = new ColorPair(
+		MajorColorNames[majorIndex],
+		MinorColorNames[minorIndex]
+	);
 	return pair;
 }
-
-function getPairNumberFromColor(pair)
-        {
-        let majorIndex = -1;
-        for (let i = 0; i < MajorColorNames.length; i++)
-            {
-                if (MajorColorNames[i] == pair.majorColor)
-                {
-                    majorIndex = i;
-                    break;
-                }
-            }
-
-        let minorIndex = -1;
-            for (let i = 0; i < MinorColorNames.length; i++)
-            {
-                if (MinorColorNames[i] == pair.minorColor)
-                {
-                    minorIndex = i;
-                    break;
-                }
-            }
-    
-        if (majorIndex == -1 || minorIndex == -1)
-            {
-                throw `Unknown Colors:${pair.toString()}`;
-            }
-
-        return (majorIndex * MinorColorNames.length) + (minorIndex + 1);
-	 }
-	 
-	 function test(){
-			pairNumber = 4;
-            let testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "WHITE");
-			console.assert(testPair1.minorColor == "BROWN");
-
-			pairNumber = 5;
-            testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "WHITE");
-			console.assert(testPair1.minorColor == "SLATEGRAY");
-			 
-			pairNumber = 23;
-            testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "RED");
-			console.assert(testPair1.minorColor == "GREEN");
-			 
-            let testPair2 = new ColorPair();
-			testPair2.majorColor="YELLOW";
-			testPair2. minorColor ="GREEN";
-            pairNumber =getPairNumberFromColor(testPair2);
-            console.log(`[In]Colors: ${testPair2}, [Out] PairNumber: ${pairNumber}`);
-			console.assert(pairNumber==18);
-
-			testPair2 = new ColorPair();
-			testPair2.majorColor="RED";
-			testPair2. minorColor ="BLUE";
-            pairNumber =getPairNumberFromColor(testPair2);
-            console.log(`[In]Colors: ${testPair2}, [Out] PairNumber: ${pairNumber}`);
-			console.assert(pairNumber==6);
+/** This function receives color pair and returns the color number for identify wires in telecommunications cables
+ * @param Color Pair
+ * @returns pairNumber
+ */
+function getPairNumberFromColor(pair) {
+	let majorIndex = -1;
+	for (let i = 0; i < MajorColorNames.length; i++) {
+		if (MajorColorNames[i] == pair.majorColor) {
+			majorIndex = i;
+			break;
+		}
+	}
+	let minorIndex = -1;
```

Comment:
This pattern repeats twice for major and minor: Pattern of initializing to -1, iterating through the elements, and comparison. Try moving this functionality into a function and call in two places.

Reply:
Added